lib/src/components/NxToast/__tests__/NxToastContainer.test.tsx

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+
+import React from 'react';
+import { rtlRenderElement, rtlRender } from '../../../__testutils__/rtlUtils';
+import { screen, render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import NxToastContainer from '../NxToastContainer';
+import { NxToastContainerProps } from '../types';
+import NxToast from '../NxToast';
+
+import NxAlert from '../../NxAlert/NxAlert';
+import { faEye } from '@fortawesome/free-solid-svg-icons';
+
+describe('NxToastContainer', function() {
+  const simpleProps: NxToastContainerProps = {
+    children:
+  <>
+    <NxToast onClose={()=>{}}>
+      <NxAlert icon={faEye}>This is an Alert</NxAlert>
+    </NxToast>
+    <NxToast onClose={()=>{}}>
+      <NxAlert icon={faEye}>This is an Alert</NxAlert>
+    </NxToast>
+    <NxToast onClose={()=>{}}>
+      <NxAlert icon={faEye}>This is an Alert</NxAlert>
+    </NxToast>
+  </>
+  };
+
+  const quickRender = rtlRender(NxToastContainer, simpleProps);
+  const renderEl = rtlRenderElement(NxToastContainer, simpleProps);
+
+  it('renders any NxToasts passed into it', function() {
+    renderEl();
+
+    const alerts = screen.getAllByText('This is an Alert');
+
+    expect(alerts.length).toBe(3);
+  });
+
+  it('sets focus to last toast when first toast is closed', async function() {
+    const user = userEvent.setup();
+    quickRender();
+
+    const closeBtns = screen.getAllByRole('button', {name: 'Close'});
+
+    await user.click(closeBtns[0]);
+
+    expect(closeBtns[2]).toHaveFocus();
+  });
+
+  it('sets focus to the next last toast when the last toast is closed', async function() {
+    const user = userEvent.setup();
+    quickRender();
+
+    const closeBtns = screen.getAllByRole('button', {name: 'Close'});
+
+    await user.click(closeBtns[2]);
+
+    expect(closeBtns[1]).toHaveFocus();
+  });
+
+  it('sets focus to previous focused element when the last remaining toast is closed',
+      async function() {
+        const Component = () => {
+          return (
+            <>
+              <button type="button">Focus Me</button>
+              <NxToastContainer>
+                <NxToast onClose={()=>{}}>
+                  <NxAlert icon={faEye}>This is an Alert</NxAlert>
+                </NxToast>
+              </NxToastContainer>
+            </>
+          );
+        };
+
+        render(<Component />);
+
+        const prevFocusBtn = screen.getByRole('button', {name: 'Focus Me'});
+        const closeBtn = screen.getByRole('button', {name: 'Close'});
+
+        prevFocusBtn.focus();
+
+        await userEvent.click(closeBtn);
```

Comment:
I'm not certain that `userEvent.click` updates the focus to `closeBtn` when it clicks it.  To be safe, we should probably explicitly focus `closeBtn` before clicking it, so that we're certain that focus is being put _back_ on `preFocusBtn` rather than having never left it.

Reply:
> I'm not certain that `userEvent.click` updates the focus to `closeBtn` when it clicks it. To be safe, we should probably explicitly focus `closeBtn` before clicking it, so that we're certain that focus is being put _back_ on `preFocusBtn` rather than having never left it.

Added in an explicit focus on the closeBtn here: [d20ef45](https://github.com/sonatype/sonatype-react-shared-components/pull/688/commits/d20ef45560339e610fd0bd7686c84637e44d9523)